test/contrib/gcs_test.py

```diff
@@ -58,10 +67,13 @@ def setUp(self):
 
             ATTEMPTED_BUCKET_CREATE = True
 
-        for item in self.client.listdir(bucket_url('')):
-            self.client.remove(item)
+        self.client.mkdir(bucket_url(''))
 
+    def tearDown(self):
+        self.client.remove(bucket_url(''), recursive=True)
```

Comment:
I think in some cases (e.g. KeyboardInterrupt or kill -9) it might not. (Internally at Vine we treat tearDown as "best-effort" and addCleanup as "yes this will execute". The root cause of that was that any failure in setUp might cause tearDown to subsequently fail. We tend to favor addCleanup in setUp instead. This probably doesn't apply here, but it's proven a nice rule of thumb :))